src/operator/tensor/elemwise_unary_op_basic.cc

```diff
@@ -1016,7 +1016,28 @@ The storage type of ``log2`` output is always dense
 .set_attr<nnvm::FGradient>("FGradient", ElemwiseGradUseIn{"_backward_log2"});
 
 MXNET_OPERATOR_REGISTER_BINARY_WITH_SPARSE_CPU_DR(_backward_log,
-                                                  unary_bwd<mshadow_op::log_grad>);
+                                                  unary_bwd<mshadow_op::log_grad>)
```

Comment:
I don't know much about this library but,

I believe it would be better to have gradients defined for  existing backward, instead of a differentiable gradient (relying on  autograd machinery) at least on `ops`  where backward is not trivial. It will allow to use existing optimised fused kernels and make sure there is no regression in the backward.

Note: `log` is relatively trivial (single reciprocal). But maybe we may see a performance regression for `sigmoid`, if we do it by relying on autograd machinery instead of the existing `_backward_sigmoid`.